src/commands/ext-dev-deprecate.ts

```diff
@@ -4,77 +4,147 @@ import * as semver from "semver";
+      if (options.delete) {
+        return deleteExt(ref, versionPredicate, options);
+      } else {
+        return deprecate(ref, versionPredicate, options);
+      }
     }
+  );
+
+async function deprecate(
+  extensionRef: refs.Ref,
+  versionPredicate: string,
+  options: ExtDevDeprecateOptions
+) {
+  const { publisherId, extensionId, version } = extensionRef;
+  if (version) {
+    throw new FirebaseError(
+      `The input extension reference must be of the format ${clc.bold(
+        "<publisherId>/<extensionId>"
+      )}. Version should be supplied in the version predicate argument.`
+    );
+  }
+  if (!publisherId || !extensionId) {
+    throw new FirebaseError(
+      `Error parsing publisher ID and extension ID from extension reference '${clc.bold(
+        refs.toExtensionRef(extensionRef)
+      )}'. Please use the format '${clc.bold("<publisherId>/<extensionId>")}'.`
+    );
+  }
+
+  // Error out if --delete and version predicate is provided
+  let filter = "";
+  if (versionPredicate) {
     const { comparator, targetSemVer } = parseVersionPredicate(versionPredicate);
-    const filter = `id${comparator}"${targetSemVer}"`;
-    const extensionVersions = await listExtensionVersions(extensionRef, filter);
-    const filteredExtensionVersions = extensionVersions
-      .sort((ev1, ev2) => {
-        return -semver.compare(ev1.spec.version, ev2.spec.version);
-      })
-      .filter((extensionVersion) => {
-        if (extensionVersion.state === "DEPRECATED" && !options.force) {
-          return false;
-        }
-        const message =
-          extensionVersion.state === "DEPRECATED" ? ", will overwrite deprecation message" : "";
-        utils.logLabeledBullet(extensionVersion.ref, extensionVersion.state + message);
-        return true;
-      });
-    if (filteredExtensionVersions.length > 0) {
-      if (!options.force) {
-        const confirmMessage =
-          "You are about to deprecate these extension version(s). Do you wish to continue?";
-        const consent = await promptOnce({
-          type: "confirm",
-          message: confirmMessage,
-          default: false,
-        });
-        if (!consent) {
-          throw new FirebaseError("Deprecation canceled.");
-        }
+    filter = `id${comparator}"${targetSemVer}"`;
+  }
+  const extensionVersions = await listExtensionVersions(refs.toExtensionRef(extensionRef), filter);
+  const filteredExtensionVersions = extensionVersions
+    .sort((ev1, ev2) => {
+      return -semver.compare(ev1.spec.version, ev2.spec.version);
+    })
+    .filter((extensionVersion) => {
+      if (extensionVersion.state === "DEPRECATED" && !options.force) {
+        return false;
       }
-    } else {
-      throw new FirebaseError("No extension versions matched the version predicate.");
+      const message =
+        extensionVersion.state === "DEPRECATED" ? ", will overwrite deprecation message" : "";
+      // TODO: Clean up usage of extensionVersion.state
```

Comment:
Curious, why?

Reply:
Clarified this comment a bit - right now, it displays either "DEPRECATED" or "PUBLISHED", and the later does not align with our current terminology.